ledger-api/grpc-definitions/com/daml/ledger/api/v1/error.proto

```diff
@@ -0,0 +1,126 @@
+//    HTTP/2 header in the request. If an application cannot read the error details from the trailers,
+//    e.g. because some middleware drops them, it can obtain the details by parsing the message.
+//    This flag does not affect asynchronous errors reported as part of completions.
+//
+//    TODO(v2) Evaluate whether we can implement this as a gRPC interceptor that listens for the header
+//     and converts the rich error details into JSON. Otherwise this might not work for
+//     errors coming from the underlying gRPC protocol.
+//     If this doesn't work, we can alternatively add a flag to all RPC requests.
+//
+//    TODO(v2) We shoule use the same JSON encoding here as what we get from our JSON API server.
+//     We want to keep the human-readable message in one of the JSON fields.
+//
+// 4. For now, only ``ErrorInfo`` is used to communicate error details.
+//    Other forms of error details may be added in the future though.
+//
+// Common error codes and details for all services
+// ***********************************************
+//
+// Individual services may use the error codes below for other cases too.
+// See the documentation of the service for details.
+//
+// ``UNAUTHENTICATED``:
+// - The request does not include a valid access token.
+//
+// ``PERMISSION_DENIED``:
+// - The claims in the token are insufficient to perform a given operation.
+//
+// ``NOT_FOUND``:
+// - The request does not include a valid ledger id.
+//
+// ``INVALID_ARGUMENT``:
+// - The payload is malformed or is missing required fields.
+//
+// ``RESOURCE_EXHAUSTED``:
+// - The participant or the ledger is overloaded. Clients should back off exponentially and retry.
+//
+// ``DEADLINE_EXCEEDED``:
+// - The server failed to produce a result within the given timeouts.
+//   For requests that change the ledger state, clients MUST assume neither
+//   that the request succeeded nor that it was rejected.
+//
+// Error codes for command submissions
+// ***********************************
+//
+// TODO(v2) Define test cases for these errors and enforce these error rules via the ledger API test tool
+//
+// This applies to the services in ``command_service.proto`` and ``command_submission_service.proto``.
+// The common error codes from above also apply here.
+// The status codes in command completions as defined in ``completion.proto`` follow the same rules.
+//
+// Some Daml ledger implementations may produce the same error codes in further situations.
+// See the documentation of the specific Daml ledger implementation.
+//
+// If the submission led to a completion event, the following error details are provided:
+// 1. If the error is reported outside of the completion event (e.g., via the command service),
+//    the metadata key ``completion_offset`` contains the corresponding completion offset.
+//    This offset can be used as a submission rank in later calls.
+// 2. If the completion event at the offset is a definite answer,
```

Comment:
@andreaslochbihler-da: the notion of a definite answer is actually not defined in the Ledger API. It would probably make sense to do that in `commands.proto`

Reply:
Given that "definite answer" is a property of a completion, I'd rather define it in `completion.proto` and add a reference here and in `commands.proto`.